### PR TITLE
oauth: extend state cookie lifetime and fix post-signup-redirection after OAuth flow

### PR DESCRIPTION
**Pull request by unknwon**
This PR includes two changes:

1. Extend OAuth state cookie lifetime from 2 minutes to 15 minutes, because 2 minutes is not generally enough if the user encounters receive OTP from emails. Fixes https://sourcegraph.atlassian.net/browse/COREAPP-215
2. Fix redirection to welcome page on cloud after OAuth flow by appending `redirect` query parameter to the OAuth authentication URL, which will be consumed by our backend upon user completing the OAuth flow. Fixes https://sourcegraph.atlassian.net/browse/COREAPP-216

Co-authored-by: Milan Freml <kopancek@users.noreply.github.com>

### CODE REVIEW

**Review by kopancek — APPROVED** [review]
LGTM 🚢

**Comment by unknwon** [comment]
Since we have code freeze, we're gonna merge it eagerly.

File: client/web/src/auth/SignUpForm.tsx [hunk 1]
```diff
@@ -275,7 +275,7 @@ export const SignUpForm: React.FunctionComponent<SignUpFormProps> = ({
                             // here because this list will not be updated during this component's lifetime.
                             <div className="mb-2" key={index}>
                                 <a
-                                    href={provider.authenticationURL}
+                                    href={`${provider.authenticationURL || ''}&redirect=/welcome`}
```
Comment (RafLeszczynski) [inline client/web/src/auth/SignUpForm.tsx:278]: @unknwon - why do we have this in two places?

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/provider.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func getStateConfig() gologin.CookieConfig {
 	cfg := gologin.CookieConfig{
 		Name:     "gitlab-state-cookie",
 		Path:     "/",
-		MaxAge:   120, // 120 seconds
+		MaxAge:   900, // 15 minutes
```
Comment (RafLeszczynski) [inline enterprise/cmd/frontend/internal/auth/gitlaboauth/provider.go:72]: @unknwon - why do we have this in two places?

Comment (unknwon) [reply enterprise/cmd/frontend/internal/auth/gitlaboauth/provider.go:72]: One for GitHub and one for GitLab.

**Pull request merged**